### ISSUE TITLE
Can this be used in standard asp.net core -non razor pages ?

### ISSUE THREAD

**Issue by giannik** [knsatishkumar/ASPNETCoreWizard#1]
I found this code while looking for a wizard to use in standard mvc controller - view architecture.
Can this be adapted to use in asp.net core  mvc sample , without razor pages. 
Is there anything that needs to be changed ?
Thank you

**Comment by knsatishkumar** [comment]
Hello yk,

I have not tried that yet without razor.

Which scripting language/frameworks are you using for the client UI?

I will be taking look into it over the weekend and keep you posted.

**Comment by giannik** [comment]
Thank you.
For client side I'm only using jquery validation and bootstrap 5.

**Comment by knsatishkumar** [comment]
Hello yk, 
I did attempt a quick & dirty way to replace RazorPages & PageModel with standard Mvc,Controller & Actions. 
Yes it can be adapted and it does work.  

Just that to make it work with Mvc , the code behind in [Pages](https://github.com/knsatishkumar/ASPNETCoreWizard/tree/master/src/Pages)/Index.cshtml has to be broken down following things have to be done  :- 
1. persisting wizard state in new data model/ view model in TempData/Session.
2. asp-page-handler like Previous , Next, Finish has to be mapped to Mvc controller actions.

Will take some more time to put these things in place and make it work with Asp.net core Mvc architecture. 
once done I shall be adding a separate Mvc sample project to demonstrate the same under this repo.

Meanwhile, thanks for letting me know UI libraries / Frameworks that you are using, was curious to understand about the context and hence asked. 
Have you found any other better options to achieve this in Mvc yet? Please share if you have tried other options.

**Comment by knsatishkumar** [comment]
An other option i found is in here explained in the blog https://bravedeveloper.com/2021/10/08/como-hacer-un-wizard-formulario-estilo-next-next-finish-con-net-core-mvc/ Much easier and simple. 
This can get complex when multiple steps are added to the wizard.

**Comment by giannik** [comment]
No, I think the article you mention with mvc wizard is probably the best approach. 
I would only replace the session object with tempdata like your example.
Thanks.
